Cards/CardBase.gd

```diff
@@ -42,8 +45,46 @@ func on_card_play(caster: Entity, base_target: Entity) -> void:
 	if caster is Player:
 		PlayerManager.player.get_energy_component().use_energy(self)	
 	
-	for effect_data: EffectData in card_effects_data:
-		var list_targets: Array[Entity] = effect_data.targeting_function.generate_target_list(base_target)
-		for current_target in list_targets:
-			effect_data.apply_effect_data(caster, current_target)
-	CardManager.on_card_action_finished.emit(self)
+	if _card_effects_queue.size() > 0:
```

Comment:
so this is basically to prevent the player from playing too many cards in a row ? Wouldn't it be better to either:
- disable cards being played when another card is played
- add animations to the queue and play them when the associated card becomes the active card ?

I think solution two would fit better with your current code

Reply:
Well ideally this code will never get reached, because our UI doesn't allow you to play a card more than once. I just added this as a check to make sure we never hit this

Reply:
you are right :+1: